### DISCUSSION TITLE
Copy subscription state

### DISCUSSION THREAD

**Discussion by mic2zar** [ravendb/ravendb#18014] [Q&A] [answered]
We have a bunch of very simple subscriptions - something like:

> from 'Courses'

When we export database, the definition of subscription is copied, but its state seems not to be copied. As a consequence, after reimporting it to a fresh database, all documents from such a collection of Courses are processed one more time. 

Are we doing something wrong? Is this a bug? Is this a feature?

**Comment by ayende** [comment]
This is by design when you are creating a subscription. The issue is about how you can tell what documents you already saw.
You can create the subscription at a particular point in time, using:

![image](https://github.com/ravendb/ravendb/assets/116915/4a8040d9-9d8f-42d0-8313-f8f77c490b05)


That will ensure that it only applies to new documents, for example

**Reply by mic2zar** [reply]
I think this solution also would not work for us or maybe I do not fully understand it. 

In our scenario we regularly copy a database for a local development (from development). We export database with all artefacts including Subscriptions. We used Oren's ideas from webinar on event sourcing and we track changes in selected collections by creating events in subscription any time the document in any of these collection changed. Here comes a challenge that on importing database each of the documents in each of these collections is now considered changed. So we create suddenly thousands of events even if there is no real change to any of these documents between the time when they were exported and imported.   

I am thinking now that maybe we should export database without subscriptions and then create them after importing (with Send Documents From Latest Document flag set to true). Or maybe you would have even a better idea (after knowing now the context) of how to address this issue (I mean: where and how to track changes of documents to create events and not to be flooded with them on every database import)?

**Reply by ayende** [reply, marked as answer]
Yes, creating the subscriptions after the import with "latest document" is the way I would recommend.
